netmanager/src/views/components/DataDisplay/DeviceView.js

```diff
@@ -218,32 +295,226 @@ export default function DeviceView() {
+      setQuantityKind(["External Temperature(\xB0C)", "External Humidity(%)"])
+    }
+    else{
+      setQuantityKind([]);
+    }
+  }
+
+  const quantityOptions = ["PM 1(µg/m3)", "PM 2.5(µg/m3)", "PM 10(µg/m3)", "External Temperature(\xB0C)", 
+  "External Temperature(\xB0F)", "External Humidity(%)", "Internal Temperature(\xB0C)", "Internal Humidity(%)", 
+  "Battery Voltage(V)", "GPS"];
+
+  const convertQuantityOptions= (myArray) => {
+    let newArray = [];
+    for (let i=0; i<myArray.length; i++){
+      if (myArray[i]=="PM 1(µg/m3)"){
+        newArray.push({"quantityKind":"PM 1", "measurementUnit":"µg/m3"})
+      }
+      else if (myArray[i]=="PM 2.5(µg/m3)"){
+        newArray.push({"quantityKind":"PM 2.5", "measurementUnit":"µg/m3"})
+      }
+      else if (myArray[i]=="PM 10(µg/m3)"){
+        newArray.push({"quantityKind":"PM 10", "measurementUnit":"µg/m3"})
+      }
+      else if (myArray[i]=="External Temperature(\xB0C)"){
+        newArray.push({"quantityKind":"External Temperature", "measurementUnit":"\xB0C"})
+      }
+      else if (myArray[i]=="External Temperature(\xB0F)"){
+        newArray.push({"quantityKind":"External Temperature", "measurementUnit":"\xB0F"})
+      }
+      else if (myArray[i]=="External Humidity(%)"){
+        newArray.push({"quantityKind":"External Humidity", "measurementUnit":"%"})
+      }
+      else if (myArray[i]=="Internal Temperature(\xB0C)"){
+        newArray.push({"quantityKind":"Internal Temperature", "measurementUnit":"\xB0C"})
+      }
+      else if (myArray[i]=="Internal Humidity(%)"){
+        newArray.push({"quantityKind":"Internal Humidity", "measurementUnit":"%"})
+      }
+      else if (myArray[i]=="Battery Voltage(V)"){
+        newArray.push({"quantityKind":"Battery Voltage", "measurementUnit":"V"})
+      }
+      else if (myArray[i]=="GPS"){
+        newArray.push({"quantityKind":"GPS", "measurementUnit":"coordinates"})
+      }
+      else{
+        newArray.push({"quantityKind":"unknown", "measurementUnit":"unknown"})
+      }
+     
+    }
+    return newArray;
+  }
+
+
+
+  const handleEditComponentOpen = () => {
+    setEditComponentOpen(true);
+  };
+  const handleEditComponentClose = () => {
+    setEditComponentOpen(false);  
+    //setComponentName('');
+  }
+  let handleEditComponentClick = (name, id, component, quantity) => {
+    return (event) => {
+      console.log(name);
+      setDeviceName(name);
+      setComponentName(id);
+      setSensorName(component);
+      setQuantityKind(quantity);
+      handleEditComponentOpen();
+
+    }
+  }
+
+  let handleEditComponentSubmit = (e) => {
+    let filter = {
+      description:sensorName, //e.g. pms5003
+      measurement: convertQuantityOptions(quantityKind),//e.g. [{"quantityKind":"humidity", "measurementUnit":"%"}]
+    }
+    console.log(JSON.stringify(filter));
+    console.log(constants.UPDATE_COMPONENT_URI+deviceName+"&comp="+componentName);
+    
+    axios.put(
```

Comment:
We could be able to set axios params using the approach below:

```
const params = {
	articleNumber: this.articleNumber,
	mode: 'bestseller',
};

const response = await axios.$get('/api/slider', { params });
```

Reply:
This will be explored in a future PR